faucet.go

```diff
@@ -313,6 +344,26 @@ func (l *lightningFaucet) zombieChanSweeper() {
 	}
 }
 
+// ongoingInvoicesSweeper is a goroutine that cleaning up the ongoingInvoices.
+//
+// NOTE: This MUST be run as a goroutine.
+func (l *lightningFaucet) ongoingInvoicesSweeper() {
+	log.Info("ongoing invoices sweeper active")
+
+	// Every two days we'll consume a new tick and perform a sweep to close out
+	// any zombies channels.
+	twoDaysTicker := time.NewTicker(time.Hour * 48)
+	for range twoDaysTicker.C {
+		log.Info("Performing ongoing invoices sweep!")
+
+		invoiceMtx.Lock()
+		for key := range ongoingInvoices {
+			delete(ongoingInvoices, key)
+		}
+		invoiceMtx.Unlock()
```

Comment:
So, this is wrong a couple of ways.

First, `ongoingInvoices` really ought to be a member of `lightningFaucet` instead of a global var.

Second, if you're trying to clean every element of the `ongoingInvoices` map, it's easier to just do `ongoingInvoices = make(map[string]ongoingInvoice)` since deleting every key from the map is equivalent to just creating a new empty map.

Third: doing this sweeping every 48h doesn't make sense for ongoing invoices: invoices are created with a 1 hour expiry so after 1 hour the invoice won't have its status updated anymore (if payment hasn't started already).

In fact, I'm not sure why you need `ongoingInvoices` at all. Can't you use the payment hash as id and query the dcrlnd for invoice info using it?

Reply:
So it makes sense in many ways.
Removing ongoingInvoices...